src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -937,10 +944,26 @@ private void ParseOption(string name, string value)
                     break;
                 case "ssl":
                     _ssl = ParseBoolean(name, value);
+                    _hasSslBeenSet = true;
                     break;
                 case "sslverifycertificate":
                     _sslVerifyCertificate = ParseBoolean(name, value);
                     break;
+                case "tls":
+                    _tls = ParseBoolean(name, value);
+                    break;
+                case "tlsallowinvalidcertificates":
```

Comment:
As we discussed with @rstam in the slack, we consider these options `tlsAllowInvalidHostnames`,  `tlsAllowInvalidCertificates` and `sslVerifyCertificate` as the same options, except the fact that `tlsAllowInvalidHostnames` and `tlsAllowInvalidCertificates` have opposite values as compared to `sslverifycertificate`.

Reply:
I think the connection string should just keep track of which options are set, and some other level should decide whether to treat them as the same or not.

Reply:
The new cases should set the new `_tlsXyz` field values... not `_sslVerifyCertificate`.